CoffeeMaker.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include "Wrappers.h"
+#include "CoffeeMaker.h"
+#include <iostream>
+
+CoffeeMaker::CoffeeMaker() : maxWaterTemp(75) 
+{ 
+    std::cout << "CoffeeMaker constructor called" << std::endl;
+}
+
+CoffeeMaker::~CoffeeMaker()
+{
+    std::cout << "CoffeeMaker destructor called" << std::endl;
+}
+
+CoffeeMaker::Cup::Cup()
+{
+    std::cout << "Cup being constructed!" << std::endl;
+    clean = true;
+    cupFull = false;
+}
+
+CoffeeMaker::Cup:: ~Cup()
+{
+    std::cout << "Cup destructor called" << std::endl;
+}
+
+void CoffeeMaker::Cup::setCupVolume(float vol)
+{
+    volume = vol;
+    std::cout << "Cup volume set: " << volume << std::endl;
+}
+
+bool CoffeeMaker::Cup::getCleanStatus() const
+{
+    std::cout << "Clean status: " << clean << std::endl;
+    return clean;  
+}
+
+void CoffeeMaker::Cup::setCupColour(std::string col)
+{
+    colour = col;
+    std::cout << "Colour set as: " << colour << std::endl;
+}
+
+bool CoffeeMaker::Cup::fillingCup(const float& vol)
+{
+    float level = 0;
+    while(level < vol)
+    {
+        std::cout << "Filling at level: " << level << std::endl;
+        level += 50;
+    }
+    cupFull = true;
+    std::cout << "Cup full" << std::endl;
+    return cupFull;
```

Comment:
`cupFull` is always true.  What is the point of a function that always returns true? 
- [ ] revise